src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingOverview.tsx

```diff
@@ -39,34 +45,86 @@ export const FoxFarmingOverview: React.FC<FoxFarmingOverviewProps> = ({
   onAccountIdChange: handleAccountIdChange,
 }) => {
   const translate = useTranslate()
+  const assets = useAppSelector(selectorState => selectorState.assets.byId)
+  const lpAsset = assets[foxEthLpAssetId]
+  const marketData = useAppSelector(selectMarketData)
   const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
   const { chainId, highestBalanceAccountAddress, contractAddress, assetReference } = query
 
-  const highestBalanceAccountId = useMemo(
-    () =>
-      highestBalanceAccountAddress
-        ? toAccountId({
-            account: highestBalanceAccountAddress,
-            chainId: ethChainId,
-          })
-        : null,
-    [highestBalanceAccountAddress],
+  const opportunityId = useMemo(
+    () => toAssetId({ chainId, assetNamespace: 'erc20', assetReference: contractAddress }),
+    [contractAddress, chainId],
+  )
+  const highestBalanceAccountIdFilter = useMemo(
+    () => ({ stakingId: opportunityId as StakingId }),
+    [opportunityId],
+  )
+  const highestBalanceAccountId = useAppSelector(state =>
+    selectHighestBalanceAccountIdByStakingId(state, highestBalanceAccountIdFilter),
+  )
+
+  const opportunityDataFilter = useMemo(
+    () => ({
+      userStakingId: serializeUserStakingId(
+        accountId!,
+        toAssetId({
+          chainId,
+          assetNamespace: 'erc20',
+          assetReference: contractAddress,
+        }) as StakingId,
+      ),
+    }),
+    [accountId, chainId, contractAddress],
+  )
+  const opportunityData = useAppSelector(state =>
+    selectUserStakingOpportunityByUserStakingId(state, opportunityDataFilter),
   )
+
+  const underlyingAssetsIcons = useMemo(
+    () => opportunityData?.underlyingAssetIds.map(assetId => assets[assetId].icon),
+    [assets, opportunityData?.underlyingAssetIds],
+  )
+
   const accountAddress = useMemo(
     () => (accountId ? fromAccountId(accountId ?? '').account : ''),
     [accountId],
   )
 
-  const filter = useMemo(
-    () => ({
-      accountAddress,
-      contractAddress,
-    }),
-    [accountAddress, contractAddress],
+  // TODO: Abstract into a selector, not relying on the LP token but rather on the sum of both underlying tokens fiat value
+  const underlyingAssetsFiatBalance = useMemo(() => {
+    const cryptoAmount = opportunityData?.stakedAmountCryptoPrecision ?? '0'
+    // TODO: add a stakingAssetId property in OpportunityMetadata ?
```

Comment:
Will leave this for a follow-up PR but if we agree on this, it will make things a lot more programmatic.
Not a deal breaker since we have opportunities known ahead-of-time but makes a lot of sense IMO to make this a proper abstraction

Reply:
https://github.com/shapeshift/web/pull/3218/commits/e3974748a9b769123cc25f0ec1418985fe29312a
Added in the follow-up surgery PR, unsure about the naming yet but confirmed this makes things a lot nicer and programmatic